source/SpinalHDL/Data types/Int.rst

```diff
@@ -338,52 +345,136 @@ To cast a ``Bool``, a ``Bits``, or an ``SInt`` into a ``UInt``, you can use ``U(
-   :widths: 2 6 2
+   :widths: 2 4 2
 
    * - Operator
      - Description
      - Return
-   * - x(y)
-     - Readbit, y : Int/UInt
+   * - x(y: Int)
+     - Static bit access of y-th bit
+     - Bool
+   * - x(x: UInt)
+     - Variable bit access of y-th bit
+     - Bool
+   * - x(offset: Int, width bits)
+     - Fixed part select of fixed width, offset is LSB index
+     - Bits(width bits)
+   * - x(offset: UInt, width bits)
+     - Variable part-select of fixed width, offset is LSB index
+     - Bits(width bits)
+   * - x(range: Range)
+     - Access a :ref:`range <range>` of bits. Ex : myBits(4 downto 2)
+     - Bits(range.size bits)
+   * - x.subdivideIn(y slices, [strict: Boolean])
+     - Subdivide x into y slices, y: Int
+     - Vec(Bits(...), y)
+   * - x.subdivideIn(y bits, [strict: Boolean])
+     - Subdivide x in multiple slices of y bits, y: Int
+     - Vec(Bits(y bit), ...)
+   * - x.msb
+     - Access most significant bit of x (highest index)
+     - Bool
+   * - x.lsb
+     - Access lowest significant bit of x (index 0)
      - Bool
-   * - x(offset, width)
-     - Read bitfield, offset: UInt, width: Int
-     - T(width bits)
-   * - x(\ :ref:`range <range>`\ )
-     - Read a range of bits. Ex : myBits(4 downto 2)
-     - T(range bits)
-   * - x(y) := z
-     - Assign bits, y : Int/UInt
+   * - mySInt.sign
+     - Access most significant (sign) bit
      - Bool
-   * - x(offset, width) := z
-     - Assign bitfield, offset: UInt, width: Int
-     - T(width bits)
-   * - x(\ :ref:`range <range>`\ ) := z
-     - Assign a range of bit. Ex : myBits(4 downto 2) := U"010"
-     - T(range bits)
+
+
+
+Some basic examples:
 
 .. code-block:: scala
 
-   // get the bit at index 4
+   // get the element at the index 4
    val myBool = myUInt(4)
+   // assign element 1
+   myUInt(1) := True
 
-   // assign bit 1 to True
-   mySInt(1) := True
+   // index dynamically
+   val index = UInt(2 bit)
+   val indexed = myUInt(index, 2 bit)
 
-   // Range
-   val myUInt_8bits = myUInt_16bits(7 downto 0)
-   val myUInt_7bits = myUInt_16bits(0 to 6)
-   val myUInt_6bits = myUInt_16Bits(0 until 6)
+   // range index
+   val myUInt_8bit = myUInt_16bit(7 downto 0)
+   val myUInt_7bit = myUInt_16bit(0 to 6)
+   val myUInt_6bit = myUInt_16bit(0 until 6)
+   // assign to myUInt_16bit(3 downto 0)
+   myUInt_8bit(3 downto 0) := myUInt_4bit
 
-   mySInt_8bits(3 downto 0) := mySInt_4bits
+   // equivalent slices, no reversing occurs
+   val a = myUInt_16bit(8 downto 4)
+   val b = myUInt_16bit(4 to 8)
+
+   // read / assign the msb / leftmost bit / x.high bit
+   val isNegative = mySInt_16bit.sign
+   myUInt_16bit.msb := False
+
+Subdivide details
+"""""""""""""""""
+
+Both overloads of ``subdivideIn`` have an optional parameter ``strict`` (i.e. ``subdivideIn(slices: SlicesCount, strict: Boolean = true)``).
+If ``strict`` is ``true`` an error will be raised if the input could not be divided evenly. If set to ``false`` the generated pieces may
```

Comment:
`evenly` maybe `exactly` ?   use of even and odd terms start to become ambiguous when the slices!=2

`... the generated pieces may vary in size as necessary.`
`... the generated pieces may vary the size of the last slice as necessary.`

Reply:
"not be divided evenly" could be instead "not be divided in equal parts"

"the generated pieces may have varying size if necessary" hmm the intended use is that the last element will be the one which "adapts"
So 
Could be : "the last elements will have the remainder"

Reply:
done and changed the last sentence to
> If set to ``false`` the last element may be smaller than the other (equal sized) elements.